From 1abc73cf345ea5167a63b4e246f0c5bdd84eaf94 Mon Sep 17 00:00:00 2001
From: Aleksander Machniak <machniak@kolabsys.com>
Date: Tue, 3 Oct 2017 10:13:10 +0200
Subject: [PATCH] Fix regression when startup() method of some "by role
 plugins" could be called when not expected, causing PHP errors

---
 plugins/kolab_auth/kolab_auth.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plugins/kolab_auth/kolab_auth.php b/plugins/kolab_auth/kolab_auth.php
index 654ccc88..a2db8fe8 100644
--- a/plugins/kolab_auth/kolab_auth.php
+++ b/plugins/kolab_auth/kolab_auth.php
@@ -331,8 +331,8 @@ class kolab_auth extends rcube_plugin
                     // Some plugins e.g. kolab_2fa use 'startup' hook to
                     // register other hooks, but when called on 'authenticate' hook
                     // we're already after 'startup', so we'll call it directly
-                    if ($loaded && $startup && ($plugin = $this->api->get_plugin($plugin))
-                        && method_exists($plugin, 'startup')
+                    if ($loaded && $startup && $plugin == 'kolab_2fa'
+                        && ($plugin = $this->api->get_plugin($plugin))
                     ) {
                         $plugin->startup(array('task' => $rcmail->task, 'action' => $rcmail->action));
                     }
-- 
2.14.1

